### PR TITLE
Fix workflow test

### PR DESCRIPTION
**Pull request by ildyria**
Comparison was with `$test` instead of `${{ github.event.inputs.branchname }}` which would obviously break as `$test` is not defined.

Also added `lychee` into gitignore because when some-one does a git clone and subsequently `docker compose up` we do not want to see those files appear. :)

### CODE REVIEW

**Comment by ildyria** [comment]
Fix is tested to work here:
https://github.com/LycheeOrg/Lychee-Docker/actions/runs/8112058445
was broken here:
https://github.com/LycheeOrg/Lychee-Docker/actions/runs/8111936708

File: .github/workflows/build_branch.yml [hunk 1]
```diff
@@ -20,7 +20,7 @@ jobs:
     steps:
       -
         name: Validate
-        run: '[ $(printf "%c" "$test") != "v" ]'
+        run: '[ $(printf "%c" "${{ github.event.inputs.branchname }}") != "v" ]'
```
Comment (d7415) [inline .github/workflows/build_branch.yml:23]: Good spot!

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1 @@
+lychee
```
Comment (d7415) [inline .gitignore:1]: This is the .gitignore, so won't effect what's added to the Docker image. Adding to the .dockerignore might be more useful, but I have no objection to either.

**Review by d7415 — APPROVED** [review]

File: .gitignore (hunk 2 above)
Comment (ildyria) [reply .gitignore:1]: yeah but I was in the mindset of people using the docker compose instead of docker build. :)

Comment (d7415) [reply .gitignore:1]: Ok. I have almost no experience there :)

**Pull request merged**